lib/rules/no-native-reassign.js

```diff
@@ -17,6 +18,8 @@ module.exports = {
             recommended: true
         },
 
+        deprecated: true,
```

Comment:
Personally, I think `deprecated` goes beyond being a documentation-level property because it has extra semantic meaning outside of docs. There's no reason documentation generators have to restrict themselves to `meta.docs`, is there? Just my two cents.